dali/test/python/test_operator_laplacian.py

```diff
@@ -0,0 +1,610 @@
+
+    def set_deriv_sizes(win_sizes, start, end):
+        if window_dim == 1:
+            win_sizes[0] = 2 * rng.integers(start, end, dtype=np.int32) + 1
+        else:
+            deriv_sizes = 2 * rng.integers(start, end, size=axes, dtype=np.int32) + 1
+            for i in range(axes):
+                win_sizes[i][i] = deriv_sizes[i]
+
+    def inner():
+        if window_dim == 0:
+            return np.int32(2 * rng.integers(1, rand_window_size_end) + 1)
+        no_smoothing = rng.choice(a=[True, False], p=[0.25, 0.75])
+        window_shape = [axes for _ in range(window_dim)]
+        if no_smoothing:
+            win_sizes = np.ones(window_shape, dtype=np.int32)
+            set_deriv_sizes(win_sizes, 1, rand_window_size_end)
+            return win_sizes
+        win_sizes = 2 * rng.integers(0, rand_window_size_end, size=window_shape, dtype=np.int32) + 1
+        set_deriv_sizes(win_sizes, 1, rand_window_size_end)
+        return win_sizes
+    return inner
+
+
+@pipeline_def
+def laplacian_per_sample_pipeline(iterator, layout, window_dim, max_window_size, axes, use_scale,
+                                  spread_scale, normalize, out_type, win_size_1d):
+    data = fn.external_source(iterator, layout=layout)
+    scale_ndim = 0
+    if window_dim is None:
+        window_size = -1
+        max_scale = 2.**(-2 * axes + 2)
+        scale = fn.random.uniform(range=[max_scale / 32, max_scale], shape=[])
+        window_arg = None
+    else:
+        assert(window_dim in [0, 1, 2])
+        window_size = fn.external_source(random_window_size(
+            window_dim, max_window_size, axes), batch=False)
+        window_arg = window_size
+        if window_dim == 2:
+            if win_size_1d:
+                window_arg = fn.reshape(window_arg, shape=[axes * axes])
+            max_scale = fn.reductions.sum(window_size, axes=1)
+            max_scale = 2.**(-max_scale + axes + 2)
+            scale_ndim = 1
+        elif window_dim == 1:
+            win_sum = fn.reductions.sum(window_size, axes=0)
+            max_scale = 2.**(-win_sum + axes + 2)
+        else:  # window_dim == 0
+            max_scale = 2.**(-(window_size - 1) * axes + 2)
+        scale_shape = fn.shapes(max_scale, dtype=types.INT32)
+        randomize_scale = fn.random.uniform(
+            range=[1 / 32, 1], shape=scale_shape)
+        scale = max_scale * randomize_scale
+    if not use_scale:
+        scale_arg = None
+        scales = -1
+    else:
+        if spread_scale:
+            assert(scale_ndim == 0)
+            scales = fn.stack(*(scale,) * axes)
+        else:
+            scales = scale
+        scale_arg = scales
+    if out_type != np.float32:
+        dtype_arg = {}
+    else:
+        dtype_arg = {"dtype": types.FLOAT}
+    edges = fn.laplacian(data, window_size=window_arg,
+                         scale=scale_arg, normalize=normalize, **dtype_arg)
+    return edges, data, window_size, scales
+
+
+def check_per_sample_laplacian(
```

Comment:
I'd add a `device` argument. We might want to reuse the test for GPU in the future

Reply:
Sure, I'll reuse what's applicable when adding gpu version, but for now I just decided not to add a parameter that is always the same. Is that OK?

Reply:
sure